### ISSUE TITLE
Performance issues displaying orders grid with a huge amount of orders in place

### ISSUE THREAD

**Issue by ioweb-gr** [PrestaShop/PrestaShop#18060]
#### Describe the bug

We have an eshop with a lot of orders. Currently over 180000. When we try to view the order grid in the backend it takes a couple minutes to open. The list displays only 20 orders at a time. 

Server specs are enough to handle it and load averages on 2 which is very low for a 12core machine
##### CPU 
vendor_id       : GenuineIntel
cpu family      : 6
model           : 158
model name      : Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz

##### HDD
Samsung NVMe 1TB

##### RAM
64GB DDR4

##### OS
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=18.04
DISTRIB_CODENAME=bionic
DISTRIB_DESCRIPTION="Ubuntu 18.04.4 LTS"

#### Expected behavior

We would expect the order grid to open within a few seconds.

#### Steps to Reproduce

1. Create 150000+ orders
2. Go to admin -> sell -> orders

**Screenshots**

#### Additional information

* PrestaShop version: 1.7.5.2
* PHP version: 7.2.28

**Comment by khouloudbelguith** [comment]
Hi @ioweb-gr,

Thank you for your report.
For your information, in the next release PS1.7.7.0, the Order page is migrated to Symfony.
We are still re-working in this page, you can follow our Symfony Migration project [here](https://github.com/PrestaShop/PrestaShop/projects/3)

Thanks!

**Comment by jocel1** [comment]
@ioweb-gr if you need help to optimize shops you can contact me

**Comment by florine2623** [comment]
Hi @ioweb-gr ,
As we have no answer from you, I close your issue.
Thanks!

**Comment by giles247** [comment]
@ioweb-gr 

Did you find a solution for this? we are running 1.7.7.3 and taking 14 seconds to load main BO Orders screen

**Comment by ioweb-gr** [comment]
@redactuk as you can see above, the issue exists, they planned to fix it by migrating the legacy code but as you said it's not fixed. Moreover the issue was closed although the issue still exists. Unfortunately we can't access orders page at all due to the amount of orders being over 200.000. The page times out completely for us even 30 minutes later. I wish I could help but...

**Comment by giles247** [comment]
@ioweb-gr 

What I did today was take the query generated on the new orders window and run it in phpMyAdmin and replicate the 14 seconds it takes to run, then modify that query to run in just 1.5 seconds. The modification that seemed to massively reduce the times was removing INNER JOIN ps_country... and instead changing the INNER JOIN ps_country_lang cl ON c.id_country to a LEFT JOIN ps_country_lang cl ON a.id_country... But I've not yet tested what consequences this has OR where I can find where this query is actually built in the revised code for 1.7.7.3. The query profilers also seems to indicate it's the SORT the accounts for 95% of the wait time. I will investigate some more.

**Comment by ioweb-gr** [comment]
@redactuk I followed the same steps with you and ended up at the same final query as you. However I didn't search to find where the query was built.

**Comment by giles247** [comment]
@khouloudbelguith can we have this issue re-opened please?

Also can someone at PS please let me know where the main BO orders query is being generated in 1.7.7.3?

**Comment by giles247** [comment]
@ioweb-gr 

I have made the following hack. This reduced my query time on 86k orders from 14.8s to 5.8s. Not great, but a big improvement.
I suspect this may not work if you support multiple languages, but as far as I can see changing to leftjoin should not return multiple records as only one country can exist per delivery address, but if someone else with far more expertise than me can tell me why this hack is a bad idea or why it will have unforeseen consequences please do!

So what I've changed is \src\Core\Grid\Query\OrderQueryBuilder.php around line 143.

I changed:
`
            ->innerJoin('a', $this->dbPrefix . 'country', 'c', 'a.id_country = c.id_country')
            ->innerJoin(
                'c',
                $this->dbPrefix . 'country_lang',
                'cl',
                'c.id_country = cl.id_country AND cl.id_lang = :context_lang_id'
            )
`

to

`
            ->leftJoin(
                'a',
                $this->dbPrefix . 'country_lang',
                'cl',
                'a.id_country = cl.id_country AND cl.id_lang = :context_lang_id'
            )
`

So basically removed the inner join on country and changed the next innerjoin to a leftjoin on a.id_country rather than c.id_country

As I said, this is a hack and should be used with caution. Ideally PS will find a better solution for shops with larger data sets.

**Comment by ioweb-gr** [comment]
I will also try it as it's a visual issue to test what happens. Ours is a multistore but doesn't hurt to try. I still wonder why the issue is not getting reopened when it's so important

**Comment by ioweb-gr** [comment]
Actually we're in an older version so it' snot possible to try your solution :(
